src/Model/FileSupport/BuyRequest/BundleDataProvider.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * ScandiPWA - Progressive Web App for Magento
+ *
+ * Copyright © Scandiweb, Inc. All rights reserved.
+ * See LICENSE for license details.
+ *
+ * @license OSL-3.0 (Open Software License ("OSL") v. 3.0)
+ * @package scandipwa/wishlist-graphql
+ * @link    https://github.com/scandipwa/wishlist-graphql
+ */
+
+declare(strict_types=1);
+
+namespace ScandiPWA\WishlistGraphQl\Model\FileSupport\BuyRequest;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Wishlist\Model\Wishlist\BuyRequest\BuyRequestDataProviderInterface;
+use Magento\Wishlist\Model\Wishlist\Data\WishlistItem;
+
+/**
+ * Data provider for bundle product buy requests
+ */
+class BundleDataProvider implements BuyRequestDataProviderInterface
+{
+    private const PROVIDER_OPTION_TYPE = 'bundle';
+
+    /**
+     * @inheritdoc
+     *
+     * @phpcs:disable Magento2.Functions.DiscouragedFunction
+     */
+    public function execute(WishlistItem $wishlistItem, ?int $productId): array
+    {
+        $bundleOptionsData = [];
+
+        foreach ($wishlistItem->getSelectedOptions() as $optionData) {
+            $optionData = \explode('/', base64_decode($optionData->getId()));
+
+            if ($this->isProviderApplicable($optionData) === false) {
+                continue;
+            }
+
+            [$optionType, $optionId, $optionValueId, $optionQuantity] = $optionData;
+
+            if ($optionType == self::PROVIDER_OPTION_TYPE) {
+                $bundleOptionsData['bundle_option'][$optionId][] = $optionValueId;
+                $bundleOptionsData['bundle_option_qty'][$optionId][] = $optionQuantity;
+            }
+        }
+        //for bundle options with custom quantity
+        foreach ($wishlistItem->getEnteredOptions() as $option) {
+            // phpcs:ignore Magento2.Functions.DiscouragedFunction
+            $optionData = \explode('/', base64_decode($option->getUid()));
+
+            if ($this->isProviderApplicable($optionData) === false) {
+                continue;
+            }
+            $this->validateInput($optionData);
+
+            [$optionType, $optionId, $optionValueId] = $optionData;
+            if ($optionType == self::PROVIDER_OPTION_TYPE) {
+                $optionQuantity = $option->getValue();
+                $bundleOptionsData['bundle_option'][$optionId][] = $optionValueId;
+                $bundleOptionsData['bundle_option_qty'][$optionId][] = $optionQuantity;
+            }
+        }
+
+        return $bundleOptionsData;
+    }
+
+    /**
+     * Validates the provided options structure
+     *
+     * @param array $optionData
+     * @throws LocalizedException
+     */
+    private function validateInput(array $optionData): void
+    {
+        if (count($optionData) !== 4) {
```

Comment:
Why exactly 4 array elements? The amount of data provided could change with time, so I would like to see a better way to validate if the data is usable or not